front/app/containers/Admin/formBuilder/components/FormFields/index.tsx

```diff
@@ -67,72 +63,167 @@ const FormFields = ({
     watch,
     formState: { errors },
   } = useFormContext();
-  const formCustomFields: IFlatCustomField[] = watch('customFields');
   const locale = useLocale();
+  const formCustomFields: IFlatCustomField[] = watch('customFields');
+
   if (isNilOrError(locale)) {
     return null;
   }
 
+  const FormFieldsContainer = styled(Box)`
+    &:hover {
+      cursor: pointer;
+    }
+  `;
+
+  const isFieldSelected = (fieldId) => {
+    return selectedFieldId === fieldId;
+  };
+
+  const getFieldBackgroundColor = (field) => {
+    if (field.input_type === 'page') {
+      return isFieldSelected(field.id) ? colors.primary : colors.background;
+    }
+    return undefined;
+  };
+
+  const getTitleColor = (field) => {
+    if (field.input_type === 'page' && isFieldSelected(field.id)) {
+      return 'white';
+    }
+    return 'grey800';
+  };
+
+  const getIndexTitleColor = (field) => {
+    if (field.input_type === 'page' && isFieldSelected(field.id)) {
+      return 'white';
+    }
+    return 'teal300';
+  };
+
+  const getIndexForTitle = (formCustomFields, field) => {
+    if (field.input_type === 'page') {
+      // Filter out questions
+      const filteredPages = formCustomFields.filter(
+        (customField) => customField.input_type === 'page'
+      );
+      // Return page title
+      return ` ${filteredPages.indexOf(field) + 1}`;
+    } else {
+      // Filter out pages
+      const filteredQuestion = formCustomFields.filter(
```

Comment:
Do we have to filter?

```
` ${formCustomFields.indexOf(field) + 1}`
```

Reply:
I thought so, since the formCustomFields also includes pages? So if we have "Page 1" as the first field, and "Question 1" as the second, the question will have index 1 and incorrectly be called "Question 2" I think?